src/util.c

```diff
@@ -2428,3 +2428,129 @@ bool nats_IsSubjectValid(const char *subject, bool wcAllowed)
     }
     return true;
 }
+
+natsStatus
+nats_marshalMetadata(natsBuffer *buf, bool comma, const char *fieldName, natsMetadata md)
+{
+    natsStatus s = NATS_OK;
+    int i;
+    const char *start = (comma ? ",\"" : "\"");
+
+    if (md.Count <= 0)
+        return NATS_OK;
+
+    IFOK(s, natsBuf_Append(buf, start, -1));
+    IFOK(s, natsBuf_Append(buf, fieldName, -1));
+    IFOK(s, natsBuf_Append(buf, "\":{", 3));
+    for (i = 0; (s == NATS_OK) && (i < md.Count); i++)
+    {
+        IFOK(s, natsBuf_AppendByte(buf, '"'));
+        IFOK(s, natsBuf_Append(buf, md.List[i * 2], -1));
+        IFOK(s, natsBuf_Append(buf, "\":\"", 3));
+        IFOK(s, natsBuf_Append(buf, md.List[i * 2 + 1], -1));
```

Comment:
0/5 I think it's best to imply (and document?) unencoded values and let the user encode. Otherwise we'd better implement compliant JSON string encoding - not the end of the world, but 1/5 as a separate PR if/when needed?

Reply:
yes that makes sense. we can always document/encode later if it becomes an issue.